src/DETHRACE/common/opponent.c

```diff
@@ -1787,33 +1794,32 @@ int MassageOpponentPosition(tOpponent_spec* pOpponent_spec, int pMassage_count)
     br_vector3 direction_v;
     LOG_TRACE("(%p, %d)", pOpponent_spec, pMassage_count);
 
-    BrVector3Set(&positive_y_vector, 0, 1, 0);
+    BrVector3Set(&positive_y_vector, 0.f, 1.f, 0.f);
     mat = &pOpponent_spec->car_spec->car_master_actor->t.t.mat;
     car_trans = &pOpponent_spec->car_spec->car_master_actor->t.t.translate.t;
     if (pMassage_count > 22) {
         return 0;
-    }
-    if (pMassage_count <= 20) {
+    } else if (pMassage_count > 20) {
+        car_trans->v[1] += (pMassage_count - 20) * 2.0f;
+        return 1;
+    } else {
         direction_v.v[0] = -pOpponent_spec->car_spec->car_master_actor->t.t.mat.m[2][0];
         direction_v.v[1] = -pOpponent_spec->car_spec->car_master_actor->t.t.mat.m[2][1];
         direction_v.v[2] = -pOpponent_spec->car_spec->car_master_actor->t.t.mat.m[2][2];
         if (pMassage_count % 4 >= 2) {
-
             BrVector3Cross(&displacement, &positive_y_vector, &direction_v);
             BrVector3Normalise(&displacement, &displacement);
             BrVector3Scale(&displacement, &displacement, (pMassage_count / 4) * 0.1f);
         } else {
-            BrVector3Normalise(&displacement, &displacement);
-            BrVector3Scale(&displacement, &displacement, (pMassage_count / 4) * 0.5f);
+            BrVector3Normalise(&direction_v, &direction_v);
```

Comment:
I think you're right - the original code here is not quite right. I think it should be this though (slightly different from what you propose:

```
BrVector3Normalise(&displacement, &direction_v);
BrVector3Scale(&displacement, &displacement, (pMassage_count / 4) * 0.5f);
```